tensorflow/tools/compatibility/tf_upgrade_v2.py

```diff
@@ -1177,6 +1177,53 @@ def __init__(self):
         for name, new_name in self.symbol_renames.items()
     }
 
+  @staticmethod
+  def _extract_glimpse_transformer(parent, node, full_name, name, logs, errors):
+    def _replace_uniform_noise_node(parent, old_value):
+      """Replaces old_value with 'uniform' or 'guassian'."""
+      uniform = ast.Str(s="uniform")
+      gaussian = ast.Str(s="gaussian")
+      new_value = ast.IfExp(body=uniform, test=old_value,
+                            orelse=gaussian)
+      # This copies the prefix and suffix on old_value to new_value.
+      pasta.ast_utils.replace_child(parent, old_value, new_value)
+      ast.copy_location(new_value, old_value)
+      # Put parentheses around noise.value (and remove the old prefix/
+      # suffix, they should only be around new_value).
+      pasta.base.formatting.set(new_value, "prefix", "(")
+      pasta.base.formatting.set(new_value, "suffix", ")")
+
+    # Check if we have a uniform_noise keyword arg
+    for uniform_noise in node.keywords:
+      if uniform_noise.arg == "uniform_noise":
+        logs.append((node.lineno, node.col_offset,
+                     "Changing uniform_noise arg of tf.image.extract_glimpse "
+                     "to noise, and recomputing value. Please check this "
+                     "transformation.\n"))
+        uniform_noise.arg = "noise"
+        value = "uniform" if uniform_noise.value else "gaussian"
+        _replace_uniform_noise_node(uniform_noise, uniform_noise.value)
+        return node
+
+    # Maybe it was a positional arg
+    if len(node.args) < 5:
+      errors.append((node.lineno, node.col_offset,
```

Comment:
I don't think this should be an error, right? It's legal to omit the arg, so if it's just not there, we don't have to do anything.

Reply:
Thanks. Fixed.